api/src/controllers/users.ts

```diff
@@ -15,6 +16,54 @@ router.get('/', function (req, res, next) {
   res.json(req.session);
 });
 
+router.get('/preferences', (req, res) => {
+  if (!req.session.passport) {
+    res.json({ error: 'Must be logged in to get preferences.' });
+    return;
+  }
+
+  const userID = req.session.passport.user.id;
+
+  getDocuments(COLLECTION_NAMES.PREFERENCES, { _id: userID }).then((preferences) => {
+    if (preferences.length > 0) {
+      res.json(preferences[0]);
+    } else {
+      res.json({ error: 'No preferences found' });
+    }
+  });
+});
+
+interface UserPreferences {
+  theme?: string;
+}
+
+router.post('/preferences', async (req, res) => {
+  if (!req.session.passport) {
+    res.json({ error: 'Must be logged in to get preferences.' });
+    return;
+  }
+
+  const userID = req.session.passport.user.id;
+
+  // make user's preference doc if it doesn't exist
+  const preferencesCollection = await getCollection(COLLECTION_NAMES.PREFERENCES);
+  if ((await preferencesCollection.find({ _id: userID }).count()) == 0) {
```

Comment:
Any reason we can't do the same logic as above?
```ts
const preferences = getDocuments(COLLECTION_NAMES.PREFERENCES, { _id: userID });
if (preferences.length === 0) // add document
```

Otherwise let's still use `===` for good practice

Reply:
We actually have a `containsID` helper that seems to do this so I'll just use that instead here.

Reply:
Ok nice I looked for that cause I thought it'd be a good thing to have but didn't notice. That's definitely better!